libs/platform/src/lib/components/dynamic-page/dynamic-page-header/header/dynamic-page-header-internal.component.ts

```diff
@@ -0,0 +1,433 @@
+    @HostBinding('attr.id')
+    id: string;
+
+    /**
+     * aria label for header
+     */
+    @Input()
+    headerAriaLabel: string;
+
+    /**
+     * aria label for pin state of pin button
+     */
+    @Input()
+    pinAriaLabel: string;
+
+    /**
+     * aria label for unpin state of pin button
+     */
+    @Input()
+    unpinAriaLabel: string;
+
+    /**
+     * sets background for content to `list`, `transparent`, or `solid` background color.
+     * Default is `solid`.
+     */
+    @Input()
+    set background(backgroundType: DynamicPageBackgroundType) {
+        if (backgroundType) {
+            this._background = backgroundType;
+            this._setBackgroundStyles(backgroundType);
+        }
+    }
+
+    get background(): DynamicPageBackgroundType {
+        return this._background;
+    }
+
+    /**
+     * sets size which in turn adds corresponding padding for the size type.
+     * size can be `small`, `medium`, `large`, or `extra-large`.
+     */
+    @Input()
+    set size(sizeType: DynamicPageResponsiveSize) {
+        if (sizeType) {
+            this._size = sizeType;
+            this._setSize(sizeType);
+        }
+    }
+
+    get size(): DynamicPageResponsiveSize {
+        return this._size;
+    }
+
+    /** Collapse/Expand change event raised */
+    @Output()
+    collapseChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    /** Reference to page header content */
+    @ViewChild('headerContent')
+    headerContent: ElementRef<HTMLElement>;
+
+    /**
+     * tracking if pin button is pinned
+     */
+    _pinned = false;
+
+    /**
+     * @hidden
+     * pn/unpin aria label based on the current state
+     */
+    _pinUnpinAriaLabel: string;
+
+    /**
+     * @hidden
+     * expand/collapse aria label based on the current state
+     */
+    _expandCollapseAriaLabel: string;
+
+    /**
+     * @hidden
+     * tracking expand/collapse button
+     */
+    private _collapsed = false;
+
+    /**
+     * @hidden
+     * tracking collapsible for pinning
+     */
+    private _collapsible: boolean;
+
+    /**
+     * @hidden
+     * tracking the background value
+     */
+    private _background: DynamicPageBackgroundType;
+
+    /**
+     * @hidden
+     * tracks the size for responsive padding
+     */
+    private _size: DynamicPageResponsiveSize;
+
+    /**
+     * @hidden
+     * subscription for when toggle header is called
+     */
+    private _toggleSubscription: Subscription;
+
+    /**
+     * @hidden
+     * subscription for when expand header is called
+     */
+    private _expandSubscription: Subscription;
+
+    /**
+     * @hidden
+     * subscription for when collapse header is called
+     */
+    private _collapseSubscription: Subscription;
+
+    /** @hidden */
+    constructor(
+        private _cd: ChangeDetectorRef,
+        private _elementRef: ElementRef<HTMLElement>,
+        private _renderer: Renderer2,
+        protected _dynamicPageConfig: DynamicPageConfig,
```

Comment:
`_dynamicPageConfig` not being used anywhere in this file, can be removed.

Reply:
removed